### PR TITLE
More thoroughly document the *.{nmalloc,ndalloc,nrequests} mallctls.

### LINKED ISSUE #412: Question: nrequests's meaning
```
"stats.arenas.<i>.small.nmalloc" (uint64_t) r- [--enable-stats]
    Cumulative number of allocation requests served by small bins.
"stats.arenas.<i>.small.ndalloc" (uint64_t) r- [--enable-stats]
    Cumulative number of small objects returned to bins. 
"stats.arenas.<i>.small.nrequests" (uint64_t) r- [--enable-stats]
    Cumulative number of small allocation requests. 

bins:           size ind    allocated      nmalloc      ndalloc    nrequests      curregs      curruns regs pgs  util       nfills     nflushes      newruns       reruns
8   0         2072        19642        19383        90384          259            1  512   1 0.505         7455          351            1            0
16   1         5776        20461        20100       113537          361            2  256   1 0.705         9034          362           15          311
32   2      1183296       276345       239367      1238871        36978          292  128   1 0.989         7239         2557          448         7186
65536  44       196608           23           20           23            3
10240  33       102400          856          846         1896           10            8    2   5 0.625          449          181          316          261
12288  34     12558336         1398          376         1034         1022         1022    1   3 1              220          140         1398            0
```

I found allocated == (nmalloc-ndalloc)*size_class, but for nrequests, it is so mysterious.
As you can see:
Sometimes nrequests is equal to nmalloc.
Sometimes nrequests is greater than nmalloc.
Sometimes nrequests is much greater than nmalloc.
And sometimes nrequests is less than nmalloc.

### PR DESCRIPTION
**Pull request by jasone**
This resolves #412.

### CODE REVIEW

File: doc/jemalloc.xml.in [hunk 1]
```diff
@@ -2623,8 +2629,8 @@ struct extent_hooks_s {
           <literal>r-</literal>
           [<option>--enable-stats</option>]
         </term>
-        <listitem><para>Cumulative number of small allocation requests.
-        </para></listitem>
+        <listitem><para>Cumulative number of allocation requests satisfied by
+        all bin size classes.</para></listitem>
```
Comment (interwq) [inline doc/jemalloc.xml.in]: Should we say "including tcache" somewhere? That was the question I had when reading the doc long time ago.

Comment (jasone) [inline doc/jemalloc.xml.in]: IMO tcache isn't directly relevant here, and trying to say anything here confuses things.  We could say these are "application-originated" allocation requests, but IMO that's redundant.  Note that I expanded the ```nmalloc``` docs to help differentiate ```nrequests```; if you read the docs for both stats does it still seem confusing?

Comment (interwq) [inline doc/jemalloc.xml.in]: The addition in ```nmalloc``` does help.

I just realized, this should be "all small bin size".

Comment (jasone) [inline doc/jemalloc.xml.in]: I was confused for a bit while updating these docs about the distinction between "bin" and "small", and realized that ideally we wouldn't talk about "bins" at all and call them [slab-backed] small size classes (probably also renaming "hbins" to something else).  But, that's a non-trivial renaming that I'm reluctant to do so close to the release.

Comment (interwq) [inline doc/jemalloc.xml.in]: 👍

**Pull request merged**